### ISSUE TITLE
feature request - zstd/compression support in crane copy

### ISSUE THREAD

**Issue by kapilt** [google/go-containerregistry#1501]
I'd like to be able to crane copy to be able to switch compressions (zstd <-> gzip) when copying. there's a lot of tools that only support one or the other, and using crane as a utility to enable that is really helpful. we already use it as a utility for slicing multi-arch images when other runtimes/tools don't support multi-arch.

**Comment by jonjohnsonjr** [comment]
I don't think we would add this as part of `crane copy` because it would mutate the images (and thus digests), so it's not really a "copy". Maybe as part of `crane mutate`?

**Comment by kapilt** [comment]
Part of mutate sounds good.

**Comment by misha-ridge** [comment]
Another place where it will be useful is `crane append`.

**Comment by misha-ridge** [comment]
not fixed yet

**Comment by misha-ridge** [comment]
not fixed yet, and I don't know how to add `lifecycle/frozen`

**Comment by slonopotamus** [comment]
Not stale, there are _two_ PRs that try to address this issue.

**Comment by slonopotamus** [comment]
Weird bot.
